### DISCUSSION TITLE
No documentation on output format specified for convert_raw_image_to_svg or convert_pixels_to_svg

### DISCUSSION THREAD

**Discussion by The-Underdog** [visioncortex/vtracer#98] [Q&A]
Digging into the vtracer code for the functions here (https://github.com/visioncortex/vtracer/blob/master/cmdapp/vtracer/vtracer.pyi), there are listed two functions:

1. convert_raw_image_to_svg, which accepts a bytes object and returns a str, and
2. convert_pixels_to_svg which accepts a pixel array and size, and returns a str

Attempting to find out what the return string actually contains points to one example (https://pypi.org/project/vtracer/) which simply says:
`svg_str: str = vtracer.convert_raw_image_to_svg(input_img_bytes, img_format = 'jpg')`
and
`svg_str: str = vtracer.convert_pixels_to_svg(pixels)`

But this tells almost nothing. There's a lack of documentation on what is returned. What does the string contain? What format is the data it holds in? Is it a byte string? Is it a text string representing vector graphics? Is it a custom vtracer format or a common standard? If it's a text string, is it unicode, UTF-8, UTF-16, or something else? Is it compressed or encoded? What should the data look like if it is successful? What sort of data sizes can I expect? Are there any other example Python libraries that will accept the data as-is for further processing or rendering?

**Comment by tyt2y3** [comment]
Hi Underdog, it's a wrapper of the rust api, which can be found here https://github.com/visioncortex/vtracer/blob/master/cmdapp/src/python.rs

The returned string is the svg's xml in UTF-8 string. In case of an error, an exception will be thrown.
